Unable to migrate to non-deprecated AnnotationsTransformerBuildItem

### Describe the bug

We've recently updated Quarkus to 3.12.0 from 3.11 in a custom extension we use internally and we noticed that `io.quarkus.arc.deployment.AnnotationsTransformerBuildItem#AnnotationsTransformerBuildItem(io.quarkus.arc.processor.AnnotationsTransformer)` is deprecated and marked for removal. I've tried to use the other constructor but that is causing some build errors.

### Expected behavior

_No response_

### Actual behavior

The build crash

```text
[ERROR] Errors: 
[ERROR]   CustomTest » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: java.lang.NullPointerException: Cannot invoke "org.jboss.jandex.AnnotationTarget.kind()" because the return value of "org.jboss.jandex.AnnotationInstance.target()" is null
	at io.quarkus.arc.processor.Annotations.getAnnotations(Annotations.java:105)
	at io.quarkus.arc.processor.Annotations.getAnnotations(Annotations.java:91)
	at io.quarkus.arc.deployment.AutoProducerMethodsProcessor$1.transform(AutoProducerMethodsProcessor.java:69)
	at io.quarkus.arc.processor.AnnotationsTransformer.apply(AnnotationsTransformer.java:67)
	at org.jboss.jandex.AnnotationOverlayImpl.getAnnotationsFor(AnnotationOverlayImpl.java:241)
	at org.jboss.jandex.AnnotationOverlayImpl.annotations(AnnotationOverlayImpl.java:211)
	at io.quarkus.arc.processor.AnnotationStore.getAnnotations(AnnotationStore.java:42)
	at io.quarkus.arc.processor.BeanDeployment.findBeans(BeanDeployment.java:1116)
	at io.quarkus.arc.processor.BeanDeployment.registerBeans(BeanDeployment.java:291)
	at io.quarkus.arc.processor.BeanProcessor.registerBeans(BeanProcessor.java:153)
	at io.quarkus.arc.deployment.ArcProcessor.registerBeans(ArcProcessor.java:439)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
	at java.base/java.lang.Thread.run(Thread.java:840)
	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
```
(for the full stacktrace see: https://github.com/DSkoufis/quarkus-annotation-transformer-error-reproducer/blob/main/README.md)

### How to Reproduce?

I've prepared a reproducer project here: https://github.com/DSkoufis/quarkus-annotation-transformer-error-reproducer

There's a build step that registers some annotations (using the new class):

```java
new AnnotationsTransformerBuildItem(AnnotationTransformation.forMethods()
        .whenMethod(m -> m.name().equals(HANDLER_METHOD_NAME))
        .whenMethod(m -> m.declaringClass().interfaceNames().stream().anyMatch(HANDLERS::contains))
        .transform(t -> {
          t.add(CustomAnnotation.class);
          t.add(ActivateRequestContext.class);
        }));
```

But when I try to create a test in that extension it's failing:

```java
@RegisterExtension
static final QuarkusUnitTest config = new QuarkusUnitTest()
    .withApplicationRoot(jar -> jar.addClass(TestRequestHandler.class));
```

I've added the previous version that still works just fine and we were using: https://github.com/DSkoufis/quarkus-annotation-transformer-error-reproducer/blob/main/deployment/src/main/java/org/acme/greeting/extension/deployment/GreetingExtensionProcessor.java#L56

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17.0.12

### Quarkus version or git rev

3.12.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvn 3.9.8

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -20,7 +20,7 @@
         <bouncycastle.tls.fips.version>1.0.19</bouncycastle.tls.fips.version>
         <expressly.version>5.0.0</expressly.version>
         <findbugs.version>3.0.2</findbugs.version>
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <parsson.version>1.1.6</parsson.version>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -28,7 +28,7 @@
         <scala-plugin.version>${scala-maven-plugin.version}</scala-plugin.version>
 
         <!-- Jandex versions -->
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <jandex-gradle-plugin.version>1.0.0</jandex-gradle-plugin.version>
 
         <asciidoctorj.version>2.5.13</asciidoctorj.version>

--- a/independent-projects/arc/pom.xml
+++ b/independent-projects/arc/pom.xml
@@ -45,7 +45,7 @@
         <version.jta>2.0.1</version.jta>
         <!-- main versions -->
         <version.gizmo>1.8.0</version.gizmo>
-        <version.jandex>3.2.0</version.jandex>
+        <version.jandex>3.2.1</version.jandex>
         <version.jboss-logging>3.6.0.Final</version.jboss-logging>
         <version.mutiny>2.6.2</version.mutiny>
         <version.bridger>1.6.Final</version.bridger>

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -34,7 +34,7 @@
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <jmh.version>1.37</jmh.version>
 
         <!-- Dependency versions -->

--- a/independent-projects/junit5-virtual-threads/pom.xml
+++ b/independent-projects/junit5-virtual-threads/pom.xml
@@ -41,7 +41,7 @@
         <compiler.plugin.version>3.13.0</compiler.plugin.version>
         <enforcer.plugin.version>3.2.1</enforcer.plugin.version>
         <surefire.plugin.version>3.2.5</surefire.plugin.version>
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <formatter-maven-plugin.version>2.24.1</formatter-maven-plugin.version>
         <impsort-maven-plugin.version>1.11.0</impsort-maven-plugin.version>
 

--- a/independent-projects/qute/pom.xml
+++ b/independent-projects/qute/pom.xml
@@ -40,7 +40,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <version.junit>5.10.3</version.junit>
         <version.assertj>3.26.3</version.assertj>
-        <version.jandex>3.2.0</version.jandex>
+        <version.jandex>3.2.1</version.jandex>
         <version.gizmo>1.8.0</version.gizmo>
         <version.jboss-logging>3.6.0.Final</version.jboss-logging>
         <version.smallrye-mutiny>2.6.2</version.smallrye-mutiny>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -45,7 +45,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <!-- Versions -->
         <jakarta.enterprise.cdi-api.version>4.1.0</jakarta.enterprise.cdi-api.version>
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <bytebuddy.version>1.14.11</bytebuddy.version>
         <junit5.version>5.10.3</junit5.version>
         <maven.version>3.9.8</maven.version>

--- a/independent-projects/tools/pom.xml
+++ b/independent-projects/tools/pom.xml
@@ -57,7 +57,7 @@
         <mockito.version>5.12.0</mockito.version>
         <quarkus.version>${project.version}</quarkus.version>
         <maven-model-helper.version>37</maven-model-helper.version>
-        <jandex.version>3.2.0</jandex.version>
+        <jandex.version>3.2.1</jandex.version>
         <system-stubs-jupiter.version>2.0.2</system-stubs-jupiter.version>
         <awaitility.version>4.2.1</awaitility.version>
         <java-properties.version>0.0.7</java-properties.version>

